### DISCUSSION TITLE
Custom Data Training on Colab

### DISCUSSION THREAD

**Discussion by TalonBvV** [mindee/doctr#1388] [Q&A] [answered]
Has anyone successfully done detection training on colab?

I am looking for a version of DocTR that outputs the progress in a more simple way, just basic standard output (print).

The reason for this is due to the fact that I haven't succeeded in training on colab due to the library used to output progress (fastprogress), and if it is possible to get it working it would still be tedious.

**Comment by odulcy-mindee** [comment]
Hello @TalonBvV :wave:,

Can you describe what's the problem you got with `fastprogress` on Colab please ?

**Reply by TalonBvV** [reply]
I get this error:
File "fastprogress.py", line 177, in show
self.out.update(HTML(self.html_code))
AttributeError: 'NoneType' object has no attribute 'update'

I tried to use the branch below to fix it, this branch add's a check to see if it's a notebook or not, but it didn't work, the result is the same.
https://github.com/fastai/fastprogress/tree/b7d338e5c6dc41e3d9ea72a23dce95569eec9a9d

**Reply by odulcy-mindee** [reply]
Ok, is it possible to share with me (here or on Slack) the notebook you used please ?

**Reply by TalonBvV** [reply]
No problem, pretty basic notebook:
https://colab.research.google.com/drive/1WoUbzrT8wvB2yktPRXLdG41hXASiH-H_?usp=sharing

**Reply by TalonBvV** [reply]
Sorry, there was a typo in that notebook, please see the correct one below:
https://colab.research.google.com/drive/1zZDif8X15L24VUAp199_iDKSyjTqZQtu?usp=sharing

**Reply by odulcy-mindee** [reply]
@TalonBvV No problem. I've been able to reproduce this issue. I'm able to make the progress bar using `tqdm` instead of `fastprogress`. I'll create a new branch / PR, I'll ping you when it's ready

**Reply by TalonBvV** [reply]
You are a legend!!!

**Reply by odulcy-mindee** [reply, marked as answer]
@TalonBvV Can you try to use this branch please ? `odulcy-mindee:replace_fastprogress_with_tqdm`

**Reply by TalonBvV** [reply]
Works perfectly! thanks a million, I should be able to get my proof of concept done now.

To anyone who may have the same problem, if the branch has not been merged yet you can clone it like below:
`git clone https://github.com/odulcy-mindee/doctr.git --branch replace_fastprogress_with_tqdm`
